xbmc/interfaces/legacy/Dialog.cpp

```diff
@@ -101,7 +101,7 @@ namespace XBMCAddon
     }
 
 
-    int Dialog::select(const String& heading, const std::vector<String>& list, int autoclose, int preselect)
+    int Dialog::select(const String& heading, const std::vector<Alternative<String, const ListItem* > > & list, int autoclose, int preselect, bool useDetails)
```

Comment:
Would it make sense to make it `Alternative<vector, vector>` instead? As it is now, you're allowed to pass a lists with both string and ListItems, contrary to what the docs say.